src/megaapi_impl.cpp

```diff
@@ -23107,6 +23148,128 @@ void MegaApiImpl::sendPendingRequests()
                                                            request->getNodeHandle()));
             break;
         }
+        case MegaRequest::TYPE_BACKUP_FOLDER:
+        {
+            // validate the request
+            if (!request->getFile() || !request->getName()) { e = API_EACCESS; break; }
+
+            // get current user
+            User* u = client->ownuser();
+
+            // get handle of remote "My Backups" folder, from user attributes
+            if (!u || !u->isattrvalid(ATTR_MY_BACKUPS_FOLDER)) { e = API_EACCESS; break; }
+            const string* handleContainerStr = u->getattr(ATTR_MY_BACKUPS_FOLDER);
+            if (!handleContainerStr) { e = API_EACCESS; break; }
+
+            std::unique_ptr<TLVstore> tlvRecords(TLVstore::containerToTLVrecords(handleContainerStr, &client->key));
+            if (!tlvRecords || !tlvRecords->find("h")) { e = API_ENOENT; break; }
+
+            const string& handleStr = tlvRecords->get("h");
+            handle h = MegaApi::base64ToHandle(handleStr.c_str());
+            if (h == UNDEF) { e = API_ENOENT; break; }
+
+            // get Node of remote "My Backups" folder
+            MegaNode* myBackupsNode = getNodeByHandle(h);
+            if (!myBackupsNode) { e = API_ENOENT; break; }
+
+            // get `DEVICE_NAME`, from user attributes
+            if (!u->isattrvalid(ATTR_DEVICE_NAMES)) { e = API_EINCOMPLETE; break; }
+            const string* deviceNameContainerStr = u->getattr(ATTR_DEVICE_NAMES);
+            if (!deviceNameContainerStr) { e = API_EINCOMPLETE; break; }
+
+            tlvRecords.reset(TLVstore::containerToTLVrecords(deviceNameContainerStr, &client->key));
+            if (!tlvRecords || !tlvRecords->find("n")) { e = API_EINCOMPLETE; break; } // is 'n' the right key ??
+
+            const string& devName64Str = tlvRecords->get("n");
+            string deviceName = Base64::btoa(devName64Str);
+            if (deviceName.empty()) { e = API_EINCOMPLETE; break; }
+
+            // get 'device-id'
+            const string& deviceId = client->getDeviceid();
+            if (deviceId.empty()) { e = API_EINCOMPLETE; break; }
+
+            // loop remote children of "My Backups" folder, to find `DEVICE_NAME`-with-'device-id' node
+            const MegaNodeList* nodeList = myBackupsNode->getChildren();
+            MegaNode* deviceNameNode = nullptr;
+            for (int i = 0; i < nodeList->size(); ++i)
+            {
+                MegaNode* child = nodeList->get(i);
+
+                // get folder name
+                const char* name = child->getName();
+                bool gotName = deviceName == name;
+
+                // get 'device-id' tag
+                const char* id = child->getCustomAttr("device-id");
```

Comment:
We should add support to create the `DEVICE_NAME` folder, which needs to set such node's attribute `"device-id"` (perhaps shorter flavor, like `"dev-id"`.

Reply:
I don't understand. Either
a) There's a misunderstanding somewhere.. This is for the case when `DEVICE_NAME` existed. In which case it must have set the attribute. If `DEVICE_NAME` didn't exist, creating it is done later.
Or
b) The attribute should be `"dev-id"` instead of `"device-id"`?

Reply:
Let me try to clarify it. The user's attribute that stores the device-id:device-name must exist, otherwise EINCOMPLETE (already done in line 23189). At this point, the DEVICE-NAME folder may or may not exist. If the latter, it needs to be created (as you plan to do in the block in line 23222).
My comment was just mentioning that this part of the puzzle needs it's counterpart: in order to consume a node's attribute with the tag `dev-id` or `device-id` (I think tags are not strings, anyway), you may want to create the folder with the node's attribute first, so you know how to consume it :)

Reply:
Sorry, I still don't understand. This code loops through existing child folders, and looks for any one having that attribute. Nothing found means that `DEVICE-NAME` child folder must be created, along with its attribute.
Should this be changed to "attempt to create `DEVICE-NAME` folder first, and if that failed, it _might_ mean that the folder already existed, so attempt to read the attribute then" ?

The API used here, namely MegaNod::getCustomAttr() does take a string as an argument. Is that the wrong one, and some other taking a different type should be used? If so, where can I find the mapping between "device-id" and a numeric id?

Reply:
> _Should this be changed to "attempt to create DEVICE-NAME folder first, and if that failed, it might mean that the folder already existed, so attempt to read the attribute then" ?_
No, it doesn't make any sense. It's good as it is.
> _The API used here, namely MegaNod::getCustomAttr() does take a string as an argument. Is that the wrong one, and some other taking a different type should be used? If so, where can I find the mapping between "device-id" and a numeric id?_
Please, try to implement the node's attribute (the case below, where it's created, and you'll learn how to deal with node's attributes) and then you will understand how to retrieve a node's attribute in the code.